### ISSUE TITLE
Peer Feedback

### ISSUE THREAD

**Issue by kellylyke** [haandahl/income-experiences#6]
# Design/Code Review 3

#### Reviewer: Kelly

|Category|Criteria|Rating/Comments|
|--------|---------|---------------|
|**Project Overview**|| |
||Which planned functionality has been met? | crud, log in capability, js charts|
|| What planned functionality has not been met? |links to user profiles from stories|
||Describe the GitHub history and what it demonstrates about the project progress during the semester.| it seems like a lot of fairly concise commits |
||Describe how peer and instructor feedback/recommendations were incorporated into the project.| It seems like everything was either implemented or considered |
||Other comments/notes?| |
|**JSPs**|| |
||Evaluate the JSPs for templating, business logic, data validation, overall look and feel.| Besides my bizarre chrome issue, the jsps are clean, but could use a little more pizzazz. The validation is great, even when there were multiple errors, it warned you about both.  |
||Other comments/notes?| |
|**Java code quality**|Evaluate the code quality for the following and identify specific areas for improvement (class, method or line number) <li>single-purpose methods <li>well-structured project <li>descriptive naming of packages, classes, methods, variables <li>classes appropriately-sized  (no monster classes) <li> CPD (copy paste detection, meaning are the same lines of code repeated?) <li>are there candidates for super/subclass relationships, abstract classes, interfaces? <li>are any values hard-coded that should be in a properties file? <li>proper exception handling <li>proper error reporting to the user <li> code documentation | I know you are in the process of refactoring, which is great and brave of you. You have some unused entities that could be removed. The properties are not hard-coded. There are still some javadocs that need to be completed.|Other comments/notes?|  |
|**Logging**|Evaluate the use of logging, for example:<li>appropriate use of logging statements in the code for error logging and debugging <li>logging levels used - info, debug, error <li> no occurrences of  System.out.printlns or printStackTrace() <li> logging works on AWS deploy| I did not find any printlns/stack traces but I think overall logging could be used a little more.|Other comments/notes?| |
|**Unit Tests**|Evaluate the unit tests, for example: <li>tests are truly a unit test rather than a high level functional test <li>test data is appropriately cleaned up or handled <li> there is full coverage of methods that perform business logic or utility functions <li>redundant code is eliminated by using set up and tear down methods, i.e., @Before, @After| There could be more tests. The tests are repeatable and cleaned up. Before is used often to set up.
||Other comments/notes?| |
|**Security**|Evaluate authentication/authorization:| |
||Is it implemented correctly and working locally as well as on AWS? |It appears to work in both places.
||Other comments/notes?| |
|**Web Service/API integration**|Evaluate the service/api integration, for example: <li> Which service/api is implemented? <li>How is  error handling of the service implemented? For example, what happens if the service is not available?| There service is the bureau of labor statistics. There is a validation message, but more error handling would be good.  |
||Other comments/notes?| |
|**Independent research topic**| What is the independent research topic?|Js charts, Lucerne, Hibernate search |
||Is the independent research topic/technique implemented in the project?| Yes |
||Other comments/notes?| The charts are a great touch|
|**Deployment**| Has the application been successfully deployed to AWS?| Yes
||Is the hosted application fully functioning?| Yes |
||Other comments/notes?| |

**Comment by haandahl** [comment]
I am addressing the weird chrome issue (custom css not being used) by including inline styling for form fields that are supposed to be hidden.  I am adding more formatting and color, and including 2 gifs!

**Comment by kellylyke** [comment]
Sorry for the delay! I will just throw some things on here that SonarLint found that the other ones might not have.

This was weird and it might not be causing you problems... but it sounds like it could? 
![Screenshot from 2019-05-08 10-16-37](https://user-images.githubusercontent.com/42480491/57386980-1f351780-717b-11e9-928a-9bb74ae650a3.png)

This one just wants you to combine your catches if you can
![Screenshot from 2019-05-08 10-18-08](https://user-images.githubusercontent.com/42480491/57387057-425fc700-717b-11e9-8144-407797258b4d.png)

I am very much in this boat, too, but here's just a friendly reminder to clean out unused imports and variables
![Screenshot from 2019-05-08 10-19-16](https://user-images.githubusercontent.com/42480491/57387102-53103d00-717b-11e9-941c-71f3bec324da.png)

Not sure if you can even get away from this but just thought I'd share it
![Screenshot from 2019-05-08 10-24-34](https://user-images.githubusercontent.com/42480491/57387205-8357db80-717b-11e9-9040-4d27d5627b89.png)

**Comment by haandahl** [comment]
Thanks Kelly!

**Comment by haandahl** [comment]
I may have fixed the first screenshot error - I had not used a constructor for that instance before.

**Comment by haandahl** [comment]
I have read all the comments here, incorporated a little more logging, and removed some unused code. I added a little testing and will add more as time allows (not much left!) with an emphasis on testing the api and possibly handling errors differently.  I am keeping a couple of classes that I believe are not instantiated anywhere because they represent data tables that still exist and are annotated for Hibernate (not sure of the repercussions if I delete).  I may not get to all the code cleanup noted but I feel I learned something by reviewing it!  Thanks!
